### ISSUE TITLE
False positive E0001 report for `# type:` line

### ISSUE THREAD

**Issue by CareF** [pylint-dev/pylint#9257]
### Bug description

pylint fails to parse comment lines starting with `type:`

```python
"""a.py file"""

# type: abc

def foobar():
    """aaa"""
    return 1
```

### Command used

```shell
pylint a.py
```

### Pylint output

```python
************* Module a
a.py:4:9: E0001: Parsing failed: 'invalid syntax (<unknown>, line 4)' (syntax-error)
```

### Expected behavior

no error

### Pylint version

```shell
pylint 3.0.2
astroid 3.0.1
Python 3.11.6 (main, Oct  2 2023, 13:45:54) [Clang 15.0.0 (clang-1500.0.40.1)]
```

### OS / Environment

MacOS

**Comment by DanielNoord** [comment]
This is correct, `# type` has syntactic meaning and this can't be parsed correctly.

As such this is a duplicate of https://github.com/pylint-dev/pylint/issues/3757.
